simulate/simulate.go

```diff
@@ -336,21 +336,34 @@ func BasesToBaseExt(seq []dna.Base) []BaseExt {
 //BaseExtToBases converts a slice of BaseExt to a slice of dna.Base
 func BaseExtToBases(seq []BaseExt) []dna.Base {
 	var newSequence []dna.Base
-	var i, j int
+	newSequence = SortBaseExtBySeqPos(seq)
+	return newSequence
+}
 
-	for i = 0; i < len(seq); i++ { //counter to put things back in order
-		for j = 0; j < len(seq); j++ { //counter to check all seq bases before incrementing to look for the next base in the newSeq
-			if seq[j].SeqPos == i {
-				newSequence = append(newSequence, seq[j].Base)
-			}
-		}
-	}
-	if len(newSequence) != len(seq) {
-		log.Fatal("Cannot find order of bases")
+//SortBaseExtBySeqPos orders a string of BaseExt by seq position
+func SortBaseExtBySeqPos(unordered []BaseExt) []dna.Base{
```

Comment:
Since this is not sorting the slice, but making a new slice, maybe a name like, BaseExtSliceToBaseSlice?